rx_signal_processing/rx_dsp_chain.cu

```diff
@@ -239,8 +249,20 @@ int main(int argc, char **argv){
     }
 
       //We need to sample early to account for propagating samples through filters.
-    int64_t extra_samples = (first_stage_dm_rate * second_stage_dm_rate *
-                        filters.get_num_third_stage_taps());
+    int64_t extra_samples = (first_stage_dm_rate * second_stage_dm_rate * third_stage_dm_rate *
+                        filters.get_num_fourth_stage_taps());
+
+    if (extra_samples < (first_stage_dm_rate * second_stage_dm_rate * filters.get_num_third_stage_taps())) {
```

Comment:
Maybe expand the comment to include what the extra if statements are accounting for?